### PR TITLE
Support more symint ops

### PR DESCRIPTION
**Pull request by yitongh**

### CODE REVIEW

**Comment by yitongh** [comment]
TODO: update openxla submodule after pr https://github.com/AlibabaPAI/openxla/pull/2 is merged.
Update: openxla submodule is updated

File: test/ds/test_bounded_dynamic_ops.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+import torch
+import torch.nn.functional as F
+import torch_xla
+import torch_xla.core.xla_model as xm
+
+sys.path.insert(1, os.path.join(sys.path[0], '..'))
+import test_utils
+
+pd = torch._C._EnablePythonDispatcher()
+dev = xm.xla_device()
```
Comment (anw90) [inline test/ds/test_bounded_dynamic_ops.py]: Use all uppercase letters like XLA or DEVICE.

File: test/ds/test_bounded_dynamic_ops.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+    logits = logits.detach().to(dev)
+    logits.requires_grad = True
+    target = target.to(dev)
+    mark_dynamic(logits, [0], [50])
+    mark_dynamic(target, [0], [50])
+    xla_out = F.nll_loss(logits, target)
+    xla_out.backward()
+    xla_grad = logits.grad
+
```
Comment (anw90) [inline test/ds/test_bounded_dynamic_ops.py:315]: Do we need to call self.assertIn for xla_out and xla_grad.

Comment (anw90) [reply test/ds/test_bounded_dynamic_ops.py:315]: one more question: Will it raise an error if I set logits and target with different bounds?

File: test/ds/test_bounded_dynamic_ops.py [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+def diff_output(testcase,
+                output1,
+                output2,
+                atol=1e-3,
+                rtol=1e-5,
+                equal_nan=True):
+  if isinstance(output1, torch.Tensor):
+    testcase.assertIsInstance(output2, torch.Tensor)
+    output2_cpu = output2.detach().cpu()
+    if output2_cpu.dtype != output1.dtype:
```
Comment (anw90) [inline test/ds/test_bounded_dynamic_ops.py]: Will the dtypes of output1 and output2 be different?

File: test/ds/test_bounded_dynamic_ops.py [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+import torch_xla.core.xla_model as xm
+
+sys.path.insert(1, os.path.join(sys.path[0], '..'))
+import test_utils
+
+pd = torch._C._EnablePythonDispatcher()
+dev = xm.xla_device()
+
+
+def mark_dynamic(t, dims, bounds):
```
Comment (anw90) [inline test/ds/test_bounded_dynamic_ops.py]: Add '_' before mark_dynamic and diff_output.

File: test/ds/test_bounded_dynamic_ops.py [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+  torch_xla._XLAC._xla_mark_bounded_dynamic(t, dims, bounds)
+
+
+def diff_output(testcase,
```
Comment (anw90) [inline test/ds/test_bounded_dynamic_ops.py]: Does this function duplicated with TestBoundedDynamicOps::diff_output?

File: test/ds/test_bounded_dynamic_ops.py (hunk 1 above)
Comment (yitongh) [reply test/ds/test_bounded_dynamic_ops.py]: done

File: test/ds/test_bounded_dynamic_ops.py (hunk 2 above)
Comment (yitongh) [reply test/ds/test_bounded_dynamic_ops.py:315]: 1. xla_out is scalar tensor, add shape check for xla_grad.
2. Yes, it will raise an error because different shape (bounds) between logits and target

File: test/ds/test_bounded_dynamic_ops.py (hunk 3 above)
Comment (yitongh) [reply test/ds/test_bounded_dynamic_ops.py]: They are the same. I remove this `if` and add dtype check.

File: test/ds/test_bounded_dynamic_ops.py (hunk 4 above)
Comment (yitongh) [reply test/ds/test_bounded_dynamic_ops.py]: done

File: test/ds/test_bounded_dynamic_ops.py (hunk 5 above)
Comment (yitongh) [reply test/ds/test_bounded_dynamic_ops.py]: Yes, I have removed this function.

**Review by anw90 — APPROVED** [review]
